server/interpreter.js

```diff
@@ -2048,6 +2047,19 @@ Interpreter.prototype.executeException = function(error) {
   throw realError;
 };
 
+/**
+ * Create and push a new state onto the statestack.
+ * @param {!Object} node AST node for the state.
+ * @return {!Object} New state.
+ */
+Interpreter.prototype.pushNode_ = function(node) {
```

Comment:
Is there a reason this is a method on `Interpreter.prototype`, rather than on `stateStack`?

Is there a reason not to make scope an optional parameter here?